contrib_bots/lib/github_detail.py

```diff
@@ -0,0 +1,120 @@
+                # It's simpler to require them to be defined.
+                sys.exit('Default owner not defined')
+        else:
+            sys.exit('Default repo not defined')
+
+    def usage(self):
+        # type: () -> None
+        return ("This plugin displays details on github issues and pull requests. "
+                "To refrence an issue or pr type @github-detail then anytime in the message it's id like '#2561'."
+                "If you want to refrence an issues from a repository other than the default use"
+                "the user/repo#id form.")
+
+    def triage_message(self, message, client):
+        # type: () -> bool
+        # Check the message contains a username mention and #<number>
+        # replying to another bot
+        if (re.search("(?:@(?:\*\*)" + re.escape(client.full_name) + ").+?#\d+", message['content']) and
+                not message['sender_email'].endswith('-bot@zulip.com')):
+            return True
+        return False
+
+    def format_message(self, details):
+        # type: (Dict[Text, Union[Text, int, bool]]) -> Text
+
+        number = details['number']
+        title = details['title']
+        link = details['html_url']
+        # Truncate if longer than 200 characters.
+        description = (details['body'][:200] +
+                       '..') if len(details['body']) > 202 else details['body']
+        status = details['state'].title()
+
+        return '**{id} | {title}** `{link}` - **{status}**\n```quote\n{description}\n```'\
+            .format(id=number, title=title, link=link, status=status, description=description)
+
+    def get_details(self, owner, repo, number):
+        # type: (Text, Text, Text) -> Dict[Text, Union[Text, Int, Bool]]
+        # Gets the details of an issues or pull request
+
+        # Try to get an issue, try to get a pull if that fails
+        try:
+            r = requests.get(
+                'https://api.github.com/repos/{owner}/{repo}/issues/{id}'.format(owner=owner, repo=repo, id=number))
+        except requests.exceptions.RequestException as e:
+            print(e)
+            return
+
+        if r.status_code == 404:
+            try:
+                r = requests.get(
+                    'https://api.github.com/repos/{owner}/{repo}/pulls/{id}'.format(owner=owner, repo=repo, id=number))
+            except requests.exceptions.RequestException as e:
+                print(e)
+                return
+
+        if r.status_code != requests.codes.ok:
+            return
+
+        return self.format_message(r.json())
+
+    def set_owner_repo(self, owner, repo):
+        if owner is None:
+            owner = self.owner
+            if repo is None:
+                repo = self.repo
+        return (owner, repo)
+
+    def handle_message(self, message, client, state_handler):
+        # type: () -> None
+        # Capture owner, repo, id
+        issue_prs = re.finditer(
+            "(?:([\w-]+)?\/)?([\w-]+)?#(\d+)", message['content'])
```

Comment:
I still think that instead of creating one regex to handle all cases, you should create three separate regexes to handle particular cases.